test/Test/Data/Map.purs

```diff
@@ -448,5 +448,23 @@ mapTests = do
     let result = left <> right
     result == right
 
+  log "any"
+  quickCheck $ \(TestMap m :: TestMap SmallKey Int) ->
+    let list = M.toUnfoldable m
+    in  case head list of
+          Nothing -> true
+          Just h  -> case last list of
+            Nothing -> true
+            Just l  -> M.any (\x -> x == snd h) m && M.any (\x -> x == snd l) m
```

Comment:
Could you clarify why you wrote the test this way? It seems like the Leaf test case is never triggered. Moreover, calling head and last on the same list will both produce the same element when the list has only one element.

Reply:
I can add a case for the empty map, i think that should trigger leaf. As for the second comment i didn't want to make the code longer by making a different branch for the singleton Map, so in that case there is one check redundant but it doesn't matter.

Reply:
How can i add a test with `assertEquals` without running quickcheck, just a normal 1 case testcase?

Reply:
I think using [`quickCheckGen`](https://pursuit.purescript.org/packages/purescript-quickcheck/8.0.1/docs/Test.QuickCheck#v:quickCheckGen) would be the ideal solution here. Then you could easily test for maps that have 0, 1, 2, or many elements.

Reply:
Couldn't figure out how quickCheckGen works and i don't need to generate values since i only wanted to add an additional test case for the empty map. Throwing an exception in the same way quickcheck would do on failure.

Reply:
If you don't know how it works, then ask. This section of my learning repo covers that: https://jordanmartinez.github.io/purescript-jordans-reference-site/content/21-Hello-World/07-Testing/test/02-Quick-Check-and-Laws/index.html

Specifically, the "problems of arbitrary" page. If you still don't understand how to use that, I can push a fix.

Reply:
Ok but like i said i don't need to generate values when we can use a literal empty map

Reply:
Is it good like this? The test runs, the main implementation is fine i think.

Reply:
I took a look at this yesterday. The implementation of `any` and `anyWithIndex` are correct. Trying to write a test that hits every possible case is difficult. By that I mean:

> (1) for every possible case that would occur for the runtime representation of the underlying tree until we hit a fixpoint, (2) for every possible position that (if checking for whether a unique element exists in the map) a unique element could appear in the map, (3) verify that the property holds

I tried to write such a test yesterday before asking myself whether it was worth it. I didn't. I think what you have is good enough.

Reply:
Thanks so much !!